### ISSUE TITLE
git clone not working

### ISSUE THREAD

**Issue by ashu2016** [NSEDownload/NSEDownload#24]
Geetting below error 

C:\Learn\py> git clone -b '4.0' https://github.com/NSEDownload/NSEDownload            
Cloning into 'NSEDownload'...
remote: Enumerating objects: 1010, done.
remote: Counting objects: 100% (277/277), done.
remote: Compressing objects: 100% (183/183), done.
remote: Total 1010 (delta 124), reused 177 (delta 70), pack-reused 733
Receiving objects: 100% (1010/1010), 1.35 MiB | 6.03 MiB/s, done.
Resolving deltas: 100% (513/513), done.
error: invalid path 'docs/NSEDownload/Post_adjustment.png:Zone.Identifier'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'

**Comment by kshitij9192** [comment]
I think this is Windows specific problem.
More specifically the naming convention in Windows.

The clone/checkout works fine in Linux environment.

**Comment by sweetatyagiaz** [comment]
why its not working with pip

**Comment by scparsawar** [comment]
Yes, I too getting similar error while cloning with git bash in windows 10

@jinit24 any ways to fix this?

**Comment by jinit24** [comment]
@ashu2016 and @scparsawar,
Can you try without -b just
```bash
git clone https://github.com/NSEDownload/NSEDownload
```
It picks branch 4.0 by default

**Comment by mohammadnoorfgli** [comment]
I tried without specifying branch, still getting the error and NSEDownload directory is empty
```
(stock-env) D:\learning\stock-prediction>git clone https://github.com/NSEDownload/NSEDownload
Cloning into 'NSEDownload'...
remote: Enumerating objects: 1010, done.
remote: Counting objects: 100% (277/277), done.
remote: Compressing objects: 100% (183/183), done.
remote: Total 1010 (delta 124), reused 177 (delta 70), pack-reused 733
Receiving objects: 100% (1010/1010), 1.35 MiB | 3.42 MiB/s, done.
Resolving deltas: 100% (513/513), done.
error: invalid path 'docs/NSEDownload/Post_adjustment.png:Zone.Identifier'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'


(stock-env) D:\learning\stock-prediction>
```
Please suggest

**Comment by jinit24** [comment]
@mohammadnoorfgli,
Can you try running `git restore --source=HEAD` and see what the output is?
Also, can you try this - https://stackoverflow.com/questions/22041752/github-clone-succeeded-but-checkout-failed? Which mentions:
```bash
git config --system core.longpaths true
```

**Comment by jaideepmurkute** [comment]
I am facing same issue on a Windows machine but not on Linux. Assuming others facing the issue were using Windows. 

Issue as per the error message is in 'docs/NSEDownload/Post_adjustment.png:Zone.Identifier' filepath. 
Windows does not allow colon(:) in the filepath. Linux does. Replacing ':' with '_' resolves the issue.  

Post_adjustment.png:Zone.Identifier -> Post_adjustment.png_Zone.Identifier  
Pre_adjustment.png:Zone.Identifier -> Pre_adjustment.png_Zone.Identifier  

I have submitted PR #26.

(Note:  
I tried suggested pointers in Issue #24 and other pointers online; but they did not help.  
DID NOT HELP: 
git config --system core.longpaths true
git config --global core.protectNTFS false
)
